jobsdb/jobsdb.go

```diff
@@ -105,6 +105,10 @@ type GetQueryParamsT struct {
 	// Limit the total number of jobs.
 	// A value less than or equal to zero will return no results
 	JobsLimit int
+	// Limit the number of Datasets to query.
+	// A value less than or equal to zero will disable this limit (no limit),
+	// only values greater than zero are considered as valid limits.
+	DSLimit int
```

Comment:
let's not expose this and use the ds limit internally in jobsdb's unexported methods

Reply:
If we don't expose ds limit, we won't be able to set value in processor, router and batch router according to their respective env right? Example: https://github.com/rudderlabs/rudder-server/pull/2446/files#diff-63264a27291123a1413bf5e86a8dcd368b9214681cafd30426fafbdaa2a182faR2203

Reply:
Each component uses a different jobsdb so we may set different limits during jobsdb initialisation using another `OptsFunc`.

Reply:
The idea behind it is that jobsdb shouldn't expose anything in its public API with respect to datasets. Datasets are an internal detail, they control how jobs are stored internally, and callers should worry about them when using the API.